src/lib.rs

```diff
@@ -19,6 +19,103 @@
 //! curve. Such signatures are used extensively by the Bitcoin network
 //! and its derivatives.
 //!
+//! To minimize dependencies, some functions are feature-gated. To generate
+//! random keys or to re-randomize a context object, compile with the "rand"
+//! feature. To de/serialize objects with serde, compile with "serde".
+//!
+//! Where possible, the bindings use the Rust type system to ensure that
+//! API usage errors are impossible. For example, the library uses context
+//! objects that contain precomputation tables which are created on object
+//! construction. Since this is a slow operation (10+ milliseconds, vs ~50
+//! microseconds for typical crypto operations, on a 2.70 Ghz i7-6820HQ)
+//! the tables are optional, giving a performance boost for users who only
+//! care about signing, only care about verification, or only care about
+//! parsing. In the upstream library, if you attempt to sign a message using
+//! a context that does not support this, it will trigger an assertation
```

Comment:
assertion. I dont think assertation is a word.

Reply:
lol. fixed in both places